src/app/models/Site.ts

```diff
@@ -69,18 +69,17 @@ export class Site extends AbstractModel implements SiteInterface {
 
   toJSON() {
     // TODO Add image, latitude, longitude, timezone
-
     return {
       id: this.id,
       name: this.name,
-      description: this.description
+      description: this.description,
     };
   }
 
-  redirectPath(project: Project): string {
+  redirectPath(project?: Project): string {
     return siteMenuItem.route.format({
-      projectId: project.id,
-      siteId: this.id
+      projectId: project?.id || this.projectIds[0] || -1,
```

Comment:
this seems dangerous - there's a possibility of a `-1` appearing in a URL? better to just return null or empty string if a valid URL is not possible

Reply:
I think ill just put a check, if no project.id or this.projectIds[0], return "/BROKEN_LINK".

Reply:
No I don't think so. Then you run the risk of "/BROKEN_LINK" actually getting inserted into an href.

I'd recommend:
- throwing if you want this to be a fatal error
- or returning something falsey if you want to fail gracefully.

But returning a string with non-zero length to indicate an error in a function that returns strings is not a great idea.